### ISSUE TITLE
Problem with swarm on docker 1.12

### ISSUE THREAD

**Issue by gaurav-dalvi** [contiv/netplugin#691]
So when I set DOCKER_HOST,
I dont see anything in docker network ls command

if I dont set DOCKER_HOST,
I do see networks created by contiv under docker network ls command.

This happens when you do make demo.

```
docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: swarm/1.2.0
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 3
 (unknown): 192.168.2.11:2385
  └ Status: Pending
  └ Containers: 0
  └ Reserved CPUs: 0 / 0
  └ Reserved Memory: 0 B / 0 B
  └ Labels:
  └ Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?
  └ UpdatedAt: 2017-01-10T22:13:26Z
  └ ServerVersion:
 (unknown): 192.168.2.10:2385
  └ Status: Pending
  └ Containers: 0
  └ Reserved CPUs: 0 / 0
  └ Reserved Memory: 0 B / 0 B
  └ Labels:
  └ Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?
  └ UpdatedAt: 2017-01-10T22:13:26Z
  └ ServerVersion:
 (unknown): 192.168.2.12:2385
  └ Status: Pending
  └ Containers: 0
  └ Reserved CPUs: 0 / 0
  └ Reserved Memory: 0 B / 0 B
  └ Labels:
  └ Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?
  └ UpdatedAt: 2017-01-10T22:13:26Z
  └ ServerVersion:
Plugins:
 Volume:
 Network:
Swarm:
 NodeID:
 Is Manager: false
 Node Address:
Security Options:
Kernel Version: 3.10.0-327.22.2.el7.x86_64
Operating System: linux
Architecture: amd64
CPUs: 0
Total Memory: 0 B
Name: 8130f300c940
Docker Root Dir:
Debug Mode (client): false
Debug Mode (server): false
WARNING: No kernel memory limit support
[vagrant@netplugin-node1 netplugin]$ netctl tenant create t1
Creating tenant: t1
[vagrant@netplugin-node1 netplugin]$ netctl network create --tenant t1 --subnet=20.1.1.0/24 --gateway=20.1.1.254 -e "vlan" n1
Creating network t1:n1
[vagrant@netplugin-node1 netplugin]$ netctl group create -t t1 n1 g1
Creating EndpointGroup t1:g1
[vagrant@netplugin-node1 netplugin]$ netctl net ls -a
Tenant  Network  Nw Type  Encap type  Packet tag  Subnet       Gateway     IPv6Subnet  IPv6Gateway
------  -------  -------  ----------  ----------  -------      ------      ----------  -----------
t1      n1       data     vlan        0           20.1.1.0/24  20.1.1.254
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$ docker net ls
docker: 'net' is not a docker command.
See 'docker --help'.
[vagrant@netplugin-node1 netplugin]$ docker network
connect     create      disconnect  inspect     ls          rm
[vagrant@netplugin-node1 netplugin]$ docker network ls -a
unknown shorthand flag: 'a' in -a
See 'docker network ls --help'.
[vagrant@netplugin-node1 netplugin]$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
[vagrant@netplugin-node1 netplugin]$ unset DOCKER_HOST
[vagrant@netplugin-node1 netplugin]$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
2ef020553771        swarm:1.2.0         "/swarm join --advert"   About an hour ago   Up About an hour    2375/tcp                 swarm-agent
8130f300c940        swarm:1.2.0         "/swarm manage -H :23"   About an hour ago   Up About an hour    0.0.0.0:2375->2375/tcp   swarm-manager
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$ netctl net ls -a
Tenant  Network  Nw Type  Encap type  Packet tag  Subnet       Gateway     IPv6Subnet  IPv6Gateway
------  -------  -------  ----------  ----------  -------      ------      ----------  -----------
t1      n1       data     vlan        0           20.1.1.0/24  20.1.1.254
[vagrant@netplugin-node1 netplugin]$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
2ca762af1cd7        bridge              bridge              local
de844c01695b        g1/t1               netplugin           global
60bdf9f20df4        host                host                local
d8ab116645a3        n1/t1               netplugin           global
f3a1645dfbc0        none                null                local
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
2ca762af1cd7        bridge              bridge              local
de844c01695b        g1/t1               netplugin           global
60bdf9f20df4        host                host                local
d8ab116645a3        n1/t1               netplugin           global
f3a1645dfbc0        none                null                local
[vagrant@netplugin-node1 netplugin]$ export DOCKER_HOST=tcp://192.168.2.10:2375
[vagrant@netplugin-node1 netplugin]$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
[vagrant@netplugin-node1 netplugin]$ docker run -itd --net="g1/t1" alpine sh
docker: Error response from daemon: No healthy node available in the cluster.
See 'docker run --help'.
[vagrant@netplugin-node1 netplugin]$ docker --version
Docker version 1.12.3, build 6b644ec
[vagrant@netplugin-node1 netplugin]$
[vagrant@netplugin-node1 netplugin]$
```

**Comment by gkvijay** [comment]
@gaurav-dalvi please verify docker 1.12 with net-demo-installer

**Comment by gaurav-dalvi** [comment]
I will. But those are two different platforms.
as of now for `make demo` its broken

**Comment by gkvijay** [comment]
Yes, we showed you that it was broken in make demo. Could you please verify with net-demo-installer to see if that is also broken

**Comment by rhim** [comment]
I think that's the default behavior of docker network command with swarm - it doesn't show these "global" entities. If you want to see them, you have to access the docker host directly.

**Comment by gaurav-dalvi** [comment]
It got resolved by https://github.com/contiv/netplugin/pull/701

Port is changed from 2375 to 3375. I will close this once i verify on baremetal.

**Comment by gaurav-dalvi** [comment]
to verify on baremetal we need this change as well. 
https://github.com/contiv/ansible/pull/307

**Comment by gkvijay** [comment]
@gaurav-dalvi, this issue and fix #701 are for make demo. Please close this if its working for you. contiv/ansible#307 is for net_demo_installer
